File: src/models/dish-kind.ts
export const DishKind = {
  Entry: 0,
  Main: 1,
  Side: 2,
  Drink: 3,
  Dessert: 4,
  Fromage: 5
} as const;

export type DishKind = typeof DishKind[keyof typeof DishKind];

File: library/src/commonMain/kotlin/ink/literate/pawnote/models/DishKind.kt
package ink.literate.pawnote.models

enum class DishKind(val code: Int) {
  Entry(0),
  Main(1),
  Side(2),
  Drink(3),
  Fromage(4),
  Dessert(5);

  companion object {
    fun fromInt(code: Int) = entries.first { it.code == code }
  }
}
